api/src/main/java/io/minio/ListIncompleteUploadsArgs.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.minio;
+
+/** Argument class of @see #listIncompleteUploads(ListIncompleteUploadsArgs args). */
+public class ListIncompleteUploadsArgs extends BucketArgs {
```

Comment:
Could you add all parameters like `listObjects()`? Similarly you could check reusing/modifying existing `objectIterator` class to accommodate this listing as well.

Reply:
IMO this not feasible as there are two different API's expecting different  query parameters. 
List Incomplete upload : https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListMultipartUploads.html
```
GET /?uploads&Delimiter=Delimiter&EncodingType=EncodingType&KeyMarker=KeyMarker&MaxUploads=MaxUploads&Prefix=Prefix&UploadIdMarker=UploadIdMarker HTTP/1.1
Host: Bucket.s3.amazonaws.com
```

List Objects : https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html
```
GET /?list-type=2&ContinuationToken=ContinuationToken&Delimiter=Delimiter&EncodingType=EncodingType&FetchOwner=FetchOwner&MaxKeys=MaxKeys&Prefix=Prefix&StartAfter=StartAfter HTTP/1.1
Host: Bucket.s3.amazonaws.com
x-amz-request-payer: RequestPayer
```
As far as using object Iterator is concerned it implements Iterator of Item type, `(ObjectIterator implements Iterator<Result<Item>> )` while list incompleteUpload needs Iterator of type Upload `Iterator<Result<Upload>>` . Item and Upload the the different response classes for these api. 
IMO  `listObjects` and `listIncompleteObjects` being separate is correct.

Reply:
We would need to do little restructure to have this and it can be done later sometime.